lib/build_targets/plugins.dart

```diff
@@ -188,15 +188,20 @@ class NativePlugins extends Target {
       for (final Directory directory
           in pluginLibDirs.where((Directory d) => d.existsSync())) {
         for (final File lib in directory.listSync().whereType<File>()) {
-          final bool isSharedLib = lib.basename.endsWith('.so');
+          final bool isSharedLib =
+              lib.basename.contains(RegExp(r'(\.so)((\..*)|$)'));
           final bool isStaticLib = lib.basename.endsWith('.a');
+
           if (isSharedLib || isStaticLib) {
-            final String libName = getLibNameForFileName(lib.basename);
-            if (userLibs.contains(libName)) {
-              continue;
+            final bool isLinkableSharedLib = lib.basename.endsWith('.so');
```

Comment:
Why should `libprotobuf-lite.so.25` be excluded from `userLibs`?

Reply:
In this way it is linked to `libflutter_plugins` after being edited by this function 
```dart
String getLibNameForFileName(String name) {
  if (name.startsWith('lib')) {
    name = name.substring(3);
  }
  if (name.lastIndexOf('.') > 0) {
    name = name.substring(0, name.lastIndexOf('.'));
  }
  return name;
} 
```
the ending compiler argument looks like this: `-lprotobuf-lite.so` and it should either be converted fully to lib name or linked with `-l:` - `-l:libprotobuf-lite.so.25`
```dart
 if (isLinkableSharedLib || isStaticLib) {
   final String libName = getLibNameForFileName(lib.basename);
   if (userLibs.contains(libName)) {
     continue;
   }
   userLibs.add(libName);
}
lib.copySync(libDir.childFile(lib.basename).path);
```
this code copies all the files to the device, but only uses the ones ending with `.so` to linking

Reply:
Shouldn't the final structure of the `lib` directory be like

```
lib
├── libflutter_plugins.so
└── libprotobuf-lite.so.25
```

Do you still think the symbolic links should be included in the TPK?

Reply:
I suppose that it is not necessary to copy all the symlinks. `libprotobuf-lite.so.25` would be enough.